src/library.js

```diff
@@ -3504,11 +3503,6 @@ mergeInto(LibraryManager.library, {
 #endif
```

Comment:
Don't we lose this checking above? Before this PR, if the runtime exited we'd end up seeing ABORT and doing a return here. That won't happen in the new code.

Reply:
If fetch is being run on syncronous mode we want to skip all of that and just call the callback on the current stack, no need to do any checking since we are not going back to the event loop

Reply:
Are you saying this is a bugfix compared to before? Or that I am misreading the code and it is NFC?

Reply:
It not really bug fix, just a refactoring. 

The fact that the synchronous version of fetch no longer checks for `runtimeExited` or `ABORT is maybe an improvement, but I don't think its really possible for this effect anyone in practice.

The checking of `runtimeExited` and/or `ABORT` is there for when we re-enter the module from the event loop.  In that case we don't run any use code.  However, in the case of synchronous fetch we are not returning from anywhere (we are still deep in the application stack).   Its was just needlessly check those globals.